features/FEATURE_BLE/targets/TARGET_CORDIO/source/CordioPalSecurityManager.cpp

```diff
@@ -0,0 +1,601 @@
+    ::ble::pal::SecurityManager(),
+    _use_default_passkey(false),
+    _default_passkey(0)
+{
+
+}
+
+CordioSecurityManager::~CordioSecurityManager()
+{
+
+}
+
+////////////////////////////////////////////////////////////////////////////
+// SM lifecycle management
+//
+
+ble_error_t CordioSecurityManager::initialize()
+{
+    return BLE_ERROR_NONE;
+}
+
+ble_error_t CordioSecurityManager::terminate()
+{
+    return BLE_ERROR_NONE;
+}
+
+ble_error_t CordioSecurityManager::reset()
+{
+    return BLE_ERROR_NONE;
+}
+
+////////////////////////////////////////////////////////////////////////////
+// Resolving list management
+//
+
+uint8_t CordioSecurityManager::read_resolving_list_capacity()
+{
+    // FIXME: implement
+    return 0;
+}
+
+ble_error_t CordioSecurityManager::add_device_to_resolving_list(
+    advertising_peer_address_type_t peer_identity_address_type,
+    const address_t &peer_identity_address,
+    const irk_t &peer_irk
+) {
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+ble_error_t CordioSecurityManager::remove_device_from_resolving_list(
+    advertising_peer_address_type_t peer_identity_address_type,
+    const address_t& peer_identity_address
+) {
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+ble_error_t CordioSecurityManager::clear_resolving_list()
+{
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+////////////////////////////////////////////////////////////////////////////
+// Feature support
+//
+
+ble_error_t CordioSecurityManager::set_secure_connections_support(
+    bool enabled, bool secure_connections_only
+) {
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+ble_error_t CordioSecurityManager::get_secure_connections_support(
+    bool &enabled
+) {
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+////////////////////////////////////////////////////////////////////////////
+// Security settings
+//
+
+ble_error_t CordioSecurityManager::set_authentication_timeout(
+    connection_handle_t, uint16_t timeout_in_10ms
+) {
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+ble_error_t CordioSecurityManager::get_authentication_timeout(
+    connection_handle_t, uint16_t &timeout_in_10ms
+) {
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+ble_error_t CordioSecurityManager::slave_security_request(
+    connection_handle_t connection,
+    AuthenticationMask authentication
+) {
+    DmSecSlaveReq(connection, authentication.value());
+    return BLE_ERROR_NONE;
+}
+
+////////////////////////////////////////////////////////////////////////////
+// Encryption
+//
+
+ble_error_t CordioSecurityManager::enable_encryption(
+    connection_handle_t connection,
+    const ltk_t &ltk,
+    const rand_t &rand,
+    const ediv_t &ediv
+) {
+    dmSecLtk_t sec_ltk;
+    memcpy(sec_ltk.key, ltk.data(), ltk.size());
+    memcpy(sec_ltk.rand, rand.data(), rand.size());
+    memcpy(&sec_ltk.ediv, ediv.data(), ediv.size());
+
+    DmSecEncryptReq(
+        connection,
+        /* FIXME: Security Level */ DM_SEC_LEVEL_ENC_AUTH,
```

Comment:
It would be good to add the security encryption level as parameter.

Reply:
this has been resolved by adding the information as bools